marconi/app/Main.hs

```diff
@@ -84,6 +89,18 @@ optionsParser =
     <*> optStrParser (long "utxo-db" <> help "Path to the utxo database.")
     <*> optStrParser (long "datum-db" <> help "Path to the datum database.")
     <*> optStrParser (long "script-tx-db" <> help "Path to the script transactions' database.")
+    <*> optAddressesParser
+    where
+        optAddressesParser =
+            builtinDataAddresses <$> optStrParser (long "addresses-to-index"
+                                          <> help ( "White space separated list of addresses to index."
+                                                    <>  " i.e \"address-1 address-2 address-3 ...\"" ) )
+        builtinDataAddresses :: Maybe String -> Maybe MambaAddresses
+        builtinDataAddresses x =  x >>= traverse asAddress . words >>= NonEmpty.nonEmpty
+
+        asAddress  :: String -> Maybe (C.AddressInEra  C.ShelleyEra )
+        asAddress = deserialiseFromRawBytesHex (proxyToAsType Proxy) . C8.pack
```

Comment:
@koslambrou I think its worth while validating mamba dresses? 

1.  Cardano.Api.isKeyAddress  appropriate for address validation?
2. if an address is invalid, should we error out, end marconi?

Reply:
Doesn't `deserialiseFromRawBytesHex` already validate the addresses? Also, I think you need to use `deserialiseFromBech32` since I'm expecting Mamba to provide addresses in Bech32 format in the CLI.